clang/lib/3C/ConstraintVariables.cpp

```diff
@@ -900,13 +916,28 @@ FunctionVariableConstraint::FunctionVariableConstraint(const Type *Ty,
     // Is this a function?
     const FunctionProtoType *FT = Ty->getAs<FunctionProtoType>();
     assert(FT != nullptr);
-    RT = FT->getReturnType();
+    // If we don't have a function declaration, but the return does have an
```

Comment:
Why does having a declaration matter? (I presume by declaration you mean "prototype" ?)